src/context.rs

```diff
@@ -1258,17 +1285,133 @@ impl SeafowlContext for DefaultSeafowlContext {
+                                            }
+
+                                            // Combine scan + filter plans, execute and store
+                                            // resulting partition ids
+                                            let base_scan = table
+                                                .partition_scan_plan(
+                                                    &None,
+                                                    group,
+                                                    &[],
+                                                    None,
+                                                    self.internal_object_store
+                                                        .inner
+                                                        .clone(),
+                                                )
+                                                .await?;
+
+                                            // Wrap the base scan plan with the filter plan
+                                            let filter_plan: Arc<dyn ExecutionPlan> =
+                                                Arc::new(FilterExec::try_new(
+                                                    filter.clone(),
+                                                    base_scan,
+                                                )?);
+
+                                            final_partition_ids.extend(
+                                                self.execute_plan_to_partitions(
+                                                    &filter_plan,
+                                                )
+                                                .await?,
+                                            );
+                                        }
+
+                                        self.partition_catalog
+                                            .append_partitions_to_table(
+                                                final_partition_ids,
+                                                new_table_version_id,
+                                            )
+                                            .await?;
+                                    }
+                                    Err(error) => {
+                                        warn!(
+                                                "Failed constructing pruning statistics for table {} (version: {}) during DELETE planning: {}",
+                                                table.name, table.table_version_id, error
+                                            );
+
+                                        // TODO: Fallback to scan + filter across all partitions?
+                                        return Err(DataFusionError::Execution(
+                                            "Failed executing DELETE statement"
+                                                .to_string(),
+                                        ));
```

Comment:
Can partition pruning fail during normal operation? I remember seeing this once:

```
 2022-09-02T15:42:54.361Z WARN  seafowl::provider       > Failed pruning the partitions for expr datepart(Utf8("YEAR"), #tpep_pickup_datetime) IN ([Int64(2020), Int64(2021)]): Error during planning: Invalid argument error: must either specify a row count or at least one column
 2022-09-02T15:42:54.381Z WARN  seafowl::provider       > Failed pruning the partitions for expr datepart(Utf8("YEAR"), #tpep_pickup_datetime) IN ([Int64(2020), Int64(2021)]): Error during planning: Invalid argument error: must either specify a row count or at least one column
 2022-09-02T15:42:54.399Z WARN  seafowl::provider       > Failed pruning the partitions for expr datepart(Utf8("YEAR"), #tpep_pickup_datetime) IN ([Int64(2020), Int64(2021)]): Error during planning: Invalid argument error: must either specify a row count or at least one column
```

If it can fail, then we should definitely treat it as if we need to inspect all partitions.

Reply:
The `Failed pruning the partitions for expr` signifies that the underlying DF pruning logic, called inside `SeafowlPruningStatistics`, failed for a particular expression, but in that case we just keep trying to prune partitions for other expressions (if any).

The error match above occurs when we fail to construct `SeafowlPruningStatistics` in the first place (in theory it can happen). I'll add a fallback to filtering across all partitions.